emscripten.py

```diff
@@ -1232,80 +1232,41 @@ def finalize_output(metadata, post, funcs_js, sending, receiving, asm_setup, the
     first_in_asm = ''
     if settings['SPLIT_MEMORY']:
       if not settings['SAFE_SPLIT_MEMORY']:
-        first_in_asm += '''
-function get8(ptr) {
+        def make_get_set(name, coercer, shift):
+          access = 'HEAP{name}s[ptr >> SPLIT_MEMORY_BITS][(ptr & SPLIT_MEMORY_MASK) >> {shift}]'.format(name=name, shift=shift)
+          format_data = {
+            'name': name,
+            'coerced_value': coercer('value'),
+            'access': access,
+            'coerced_access': coercer(access),
+          }
+          getter = '''
+function get{name}(ptr) {{
```

Comment:
It can't, that would be much more immediately readable as an escape sequence
The problem is the `.format` method interpreting the brace as part of its control, whereas `\{` would help if it were a language-level escaping issue, this is a library-level interpretation issue, so no dice.
https://docs.python.org/2/library/string.html#format-string-syntax

I'm inclined to leave it, because I suspect it's a lot less weird looking if the reader is familiar with the `format` function. It's also still valid JS with the extra braces, so if it ever gets naively changed to `%s`-style in the future it'll still work. And if someone thinks "we don't need these I can remove a brace" they'll hit the `KeyError` I ran in to (because `format` then interprets the entire function body as the key). So it's ultimately pretty harmless.